cranelift/codegen/src/isa/riscv64/inst/emit.rs

```diff
@@ -2893,6 +2894,41 @@ impl MachInstEmit for Inst {
                 .emit(&[], sink, emit_info, state);
                 sink.bind_label(label_done);
             }
+            &Inst::ElfTlsGetAddr { ref symbol, rd } => {
+                let rd = allocs.next_writable(rd);
+                assert_eq!(a0(), rd.to_reg());
+                sink.add_reloc(Reloc::RiscvTlsGd, symbol, 0);
+                Inst::Auipc {
+                    rd: rd,
+                    imm: Imm20::from_bits(0),
+                }
+                .emit(&[], sink, emit_info, state);
+                Inst::AluRRImm12 {
+                    alu_op: AluOPRRI::Addi,
+                    rd: rd,
+                    rs: rd.to_reg(),
+                    imm12: Imm12::from_bits(0),
+                }
+                .emit(&[], sink, emit_info, state);
+                Inst::Call {
+                    info: Box::new(CallInfo {
+                        dest: ExternalName::LibCall(LibCall::ElfTlsGetAddr),
+                        uses: smallvec![CallArgPair {
+                            preg: a0(),
+                            vreg: a0()
+                        }],
+                        defs: smallvec![CallRetPair {
+                            vreg: Writable::from_reg(a0()),
+                            preg: a0(),
+                        }],
+                        opcode: crate::ir::Opcode::TlsValue,
+                        caller_callconv: CallConv::SystemV,
+                        callee_callconv: CallConv::SystemV,
+                        clobbers: PRegSet::empty(),
```

Comment:
This should take the clobber set from the C call conv (and remove a0)

Reply:
@bjorn3 , I forget this.

Reply:
This should do at `riscv64_get_operands`.